.github/workflows/check-for-workflow-run.js

```diff
@@ -0,0 +1,29 @@
+// Check for a successful previous run of a workflow.
+// This could make a decent reusable action itself.
+module.exports = async ({github, context, core}, workflowID, branch, sha = null) => {
+  // This API doesn't support filtering by SHA, so we just
+  // fetch the first page and scan manually.
+  // That means if the run is fairly old it may be missed,
+  // but that should be rare.
```

Comment:
I slightly worry that this could be fragile, but in the worst case it would be sufficient to re-trigger the deep tests manually, right?

Reply:
Exactly. The default page size for this API is 30 (https://docs.github.com/en/rest/actions/workflow-runs#list-workflow-runs), and if the nightly build hasn't run at all for a month than we have other problems to deal with. :)